### ISSUE TITLE
VIRUS DETECTED - W32.Obfuscated.Gen  

### ISSUE THREAD

**Issue by mmalewski** [RandomEngy/VidCoder#396]
If I download the latest stable release of VidCoder (3.16), a pop-up "Threats Identified" will appear, and it shows that the file "VidCoder-3.16.exe" is infected with a virus.

It says that the Infection is:  W32.Obfuscated.Gen

Webroot automatically quarantines the infected file.

<img width="509" alt="vidcoder-3 16_webroot_threats_identified" src="https://user-images.githubusercontent.com/133161/45412626-bc679c80-b63c-11e8-9a3d-8dc0e43ddbe3.png">

I checked the HASH for this file, and it seems to be a "known infected file" infected with W32.Obfuscated.Gen.

<img width="770" alt="threat_analysis_virus_total" src="https://user-images.githubusercontent.com/133161/45412797-313ad680-b63d-11e8-8ee0-ba2e17a90131.png">

Any ideas as to what could be causing Webroot to detect the W32.Obfuscated.Gen infection on the Vidcoder-3.16.exe  installer/file?

**Comment by mmalewski** [comment]
Also, can we please keep this "issue" open until we can figure out what exactly is causing this, and get this issue "resolved" with Webroot?  (And also leave this issue here for future reference, in case anyone else has this same exact issue in the future).

Thank-you!

**Comment by mmalewski** [comment]
Just out of curiosity, I downloaded the very latest VidCoder-4.12-Beta.exe, and ran a Webroot scan on it, and it came back clean.  I also checked the hash (for the "VidCoder-4.12-Beta.exe" file), and it came back clean as well.

<img width="508" alt="vidcoder-4 12-beta_webroot_clean_no_threats_identified" src="https://user-images.githubusercontent.com/133161/45414342-b8d61480-b640-11e8-97dd-ef947bbd9e2c.png">

So I'm going to install the very latest VidCoder-4.12-Beta.exe file (since it seems to be "clean").

I'm just not quite sure why the VidCoder-3.16.exe keeps triggering as "infected" by "W32.Obfuscated.Gen" and I'm just weary and don't want to install any file that comes up as "infected".

So I just wanted to make the developers "aware" of this "issue".

Also, please leave this "issue" here, so that if anyone does a search (in the future) they can find this "issue" and realize that maybe it's a "false positive"?  I'm not quite sure what is causing the Vidcoder-3.16.exe to show as "infected" and trigger my anti-virus software (as an "infected threat").

Just wanted to make everyone (including the developers) aware of this "issue".  

Thank-you!

P.S.  I submitted (and checked) the 4.12 Beta file hash online, and it seems to come back clean:

<img width="780" alt="threat_analysis_virus_total_beta_4 12_01" src="https://user-images.githubusercontent.com/133161/45414159-63016c80-b640-11e8-801a-5e68dd693a4a.png">
<img width="774" alt="threat_analysis_virus_total_beta_4 12_02" src="https://user-images.githubusercontent.com/133161/45414161-63016c80-b640-11e8-813b-8e5c00ae1fb2.png">

Webroot shows that the latest 4.12 Beta comes back clean:
<img width="508" alt="vidcoder-4 12-beta_webroot_clean_no_threats_identified" src="https://user-images.githubusercontent.com/133161/45414131-4fee9c80-b640-11e8-806a-c1b30625f5a6.png">

For reference, I've attached my Webroot Scan Log (which shows the Trigger/Threat) below:

[Webroot_Scan_Log_2018-09-12.txt](https://github.com/RandomEngy/VidCoder/files/2374403/Webroot_Scan_Log_2018-09-12.txt)

Just wanted to make everyone (including the developers) aware of this "issue".

The latest **VidCoder-4.12-Beta.exe** file seems to come back clean.  **(See attached Scan Log above)**

Any ideas as to what could be causing Webroot to detect the **W32.Obfuscated.Gen** infection on the latest **Vidcoder-3.16.exe** STABLE installer/file?  **(See attached Scan Log above)**
<img width="509" alt="vidcoder-3 16_webroot_threats_identified" src="https://user-images.githubusercontent.com/133161/45415506-6e09cc00-b643-11e8-9168-86c283d1be4a.png">

**Comment by spikaspika** [comment]
Sorry, but Kaspersky does not find any viruses, however, like other antiviruses. Why do you use such a specific antivirus?

**Comment by RandomEngy** [comment]
Every once in a while I get a report of VidCoder being detected as malware, just because virus detection is hard and heuristics are going to have false positives when matching against thousands of malware variants. Some AV companies allow you to submit a file and say "hey, this really isn't a virus, can you take a closer look".

**Comment by bcurran3** [comment]
VirusTotal results can be found here:
https://www.virustotal.com/en/file/3e8fd9e98a36afa9a0f51ba9d992f14d42b6762c94d13fbc1d1295c4686ad93a/analysis/

Summary:

> SHA256: | 3e8fd9e98a36afa9a0f51ba9d992f14d42b6762c94d13fbc1d1295c4686ad93a
> -- | --
> File name: | VidCoder-3.15.exe
> Detection ratio: | 0 / 67
> Analysis date: | 2018-09-12 13:21:36 UTC ( 1 hour, 12 minutes ago )

Sorry, I don't have 3.16 results. I install via Chocolatey and 3.15 is the latest version there.
https://chocolatey.org/packages/vidcoder

**Comment by mmalewski** [comment]
@bcurran3 - The VirusTotal Results (for VidCoder-3.16.exe) (Latest Stable) can be found here:
https://www.virustotal.com/#/file/c2264d8fd719a2e2f85f569e57dc33bbea8126526168868eb3140d1b49dec781/detection


@bcurran3 - The latest Stable (3.16) can be downloaded from the VidCoder.net website here:
http://vidcoder.net/

<img width="861" alt="vidcoder_download" src="https://user-images.githubusercontent.com/133161/45484408-9bc14480-b719-11e8-8faf-ac99f4e31f0a.png">

I was just pointing out that Webroot is detecting the **W32.Obfuscated.Gen** infection 
on the latest **Vidcoder-3.16.exe** STABLE installer/file.  

More than likely, it's probably a "false positive"? 

<img width="510" alt="vidcoder-3 16_webroot_threat_removal" src="https://user-images.githubusercontent.com/133161/45484573-30c43d80-b71a-11e8-9aad-70bb4acb36f4.png">


But regardless, Webroot is removing the threat anytime I try to download it, or try to install it onto my computer.  (I was just making you "aware" of the issue)

The VirusTotal Results (for [VidCoder-3.16.exe](https://github.com/RandomEngy/VidCoder/releases/download/v3.16/VidCoder-3.16.exe)) (Latest Stable) can be found here:
https://www.virustotal.com/#/file/c2264d8fd719a2e2f85f569e57dc33bbea8126526168868eb3140d1b49dec781/detection

So I am currently just using the latest **[VidCoder-4.12-Beta.exe](https://github.com/RandomEngy/VidCoder/releases/download/v4.12-beta/VidCoder-4.12-Beta.exe)** file which seems to come back clean. (See attached Scan Log above)

The VirusTotal Results (for [VidCoder-4.12-Beta.exe](https://github.com/RandomEngy/VidCoder/releases/download/v4.12-beta/VidCoder-4.12-Beta.exe)) (Latest Beta) can be found here:
https://www.virustotal.com/#/file/9f73a456433de14cc7ca03ae72e6d598c627dc632f58df4d42ffddfb2a270595/detection

**Comment by mmalewski** [comment]
@RandomEngy  - I completely understand, and if you read my post, I was simply "pointing out" that Webroot was detecting the latest stable as infected.  I was just making you "aware" of the issue.

<img width="509" alt="vidcoder-3 16_webroot_threats_identified" src="https://user-images.githubusercontent.com/133161/45486504-fd38e180-b720-11e8-85bf-f7652204323b.png">

@spikaspika - I am not here to discuss the "merits" of various vendors antivirus software.  I'm just saying on our own internal "testing" (in our lab) of a "test sample" of 40,000+ KNOWN viruses, Kapersky scored among the lowest (only detecting less than 17% of KNOWN viruses) in our lab test/sample.

As for WHY we use "Webroot" (among 4 other antivirus software) is because Webroot detected 92% of the viruses (in our lab test) and was found to be the "most effective" (and quickest) at scanning and detecting (and lowest number of false positives).

@spikaspika - As for WHY you are using Kapersky, that would a whole separate topic/discussion, but just because you CHOOSE to use an antiquated antivirus software, which [doesn't detect 83% of KNOWN viruses/infections](https://www.cnet.com/news/kaspersky-lab-russian-hacking-us-government-national-security-faq/) doesn't bode well for your "argument".  

@spikaspika  U.S. Government Employees are NOT ALLOWED to use Kapersky software (after we discovered various "backdoor trojans" in Kapersky Antivirus Software).  [Kapersky has been selling "infected software" to civilians for years](https://www.dhs.gov/news/2017/09/13/dhs-statement-issuance-binding-operational-directive-17-01).  As for why YOU are still using it, I'm not sure.  

Nobody working for the U.S. Government is allowed to use [ANY Kapersky software](https://www.dhs.gov/news/2017/09/13/dhs-statement-issuance-binding-operational-directive-17-01).  We have conducted our own independent lab tests, and our findings align with those of various independent researchers, and several U.S. Intelligence Agents.  As a result, [Kapersky software](https://www.dhs.gov/news/2017/09/13/dhs-statement-issuance-binding-operational-directive-17-01) is currently NOT deemed "safe to use" and is [no longer being used by the U.S. Government](https://www.dhs.gov/news/2017/09/13/dhs-statement-issuance-binding-operational-directive-17-01) or it's employees (or U.S. Intelligence).

@RandomEngy - As for "false positive", this is the FIRST "false positive" I have seen with Webroot, EVER.  I'm not saying that it doesn't happen, but it does seem this is the FIRST TIME that I've seen a "false positive" with Webroot.  I was simply here to make you "aware" of the issue.  I was only pointing the "issue" out to you.  It seems to be a very rare occurrence, but I was just making the "issue" known, and pointing it out to you.

@spikaspika - As for [Kapersky not detecting infected files](https://www.cnet.com/news/kaspersky-lab-russian-hacking-us-government-national-security-faq/), that's NORMAL, and [83% of the time Kapersky doesn't detect infected files](https://www.cnet.com/news/kaspersky-lab-russian-hacking-us-government-national-security-faq/).  As for WHY the U.S. Government (and it's employees) and [U.S. Intelligence Agents don't use Kapersky](https://www.wsj.com/articles/russian-hackers-stole-nsa-data-on-u-s-cyber-defense-1507222108)... that's a whole separate discussion.

@spikaspika - I'm not here to have a discussion about various vendors antivirus software.  

@RandomEngy - I was only here to point out that Webroot is detecting [VidCoder-3.16.exe](https://github.com/RandomEngy/VidCoder/releases/download/v3.16/VidCoder-3.16.exe) (Latest Stable) as "infected" with the **W32.Obfuscated.Gen** infection on the latest [Vidcoder-3.16.exe](https://github.com/RandomEngy/VidCoder/releases/download/v3.16/VidCoder-3.16.exe) STABLE installer/file.

I supposed this is an "issue" you may want to take up with Webroot.  I was just making you "aware" of the issue, and letting you know that Webroot is "quarantining" your installs of the [VidCoder-3.16.exe](https://github.com/RandomEngy/VidCoder/releases/download/v3.16/VidCoder-3.16.exe) (Latest Stable).

**Comment by spikaspika** [comment]
To "mmalewski".
All this looks like a not very clever advertisement WebRoot. Sorry, I do not want to discuss any more.

**Comment by RandomEngy** [comment]
Since you didn't want to, I reported it as a false positive.

**Comment by mmalewski** [comment]
@RandomEngy - I do understand that "False Positives" do happen.  I do believe that this is more than likely a "False Positive", but I was just making you "aware" of the issue.

I'm not here to be "attacked" by @spikaspika  just because I made an "issue" known.

Nor am I here to discuss [Kapersky software](https://www.dhs.gov/news/2017/09/13/dhs-statement-issuance-binding-operational-directive-17-01) (or any of the [Known Security Vulnerabilities of Kapersky software](https://www.nytimes.com/2017/10/10/technology/kaspersky-lab-israel-russia-hacking.html)).

I'm just pointing out that Webroot is detecting that [VidCoder-3.16.exe](https://github.com/RandomEngy/VidCoder/releases/download/v4.12-beta/VidCoder-4.12-Beta.exe) (Latest Stable) as "infected" with the **W32.Obfuscated.Gen** infection on the latest Vidcoder-3.16.exe STABLE installer/file.

I already opened a Support Ticket with Webroot, and I was just suggesting that you (as the developer) do the same.  Thank-you for understanding.
<img width="509" alt="vidcoder-3 16_webroot_threats_identified" src="https://user-images.githubusercontent.com/133161/45491984-b6eb7e80-b730-11e8-9d3a-89a5f7b22380.png">

I was just pointing out the "issue" with Webroot quarantining the installer file.

**Comment by mmalewski** [comment]
@RandomEngy - I completely understand, and if you read my post, I was simply "pointing out" that Webroot was detecting the latest stable as infected. I was just making you "aware" of the issue.

I do agree that it's probably a "False Positive" and I had already opened a "Trouble Ticket" with Webroot (earlier this morning, before you responded).  I didn't want to open a "Trouble Ticket" without first discussing it with you.

I had already opened a Trouble Ticket (over an hour ago) prior to your last response.

<img width="814" alt="webroot_trouble_ticket" src="https://user-images.githubusercontent.com/133161/45493037-70e3ea00-b733-11e8-936a-f3adf9242e51.png">

**Comment by mmalewski** [comment]
@RandomEngy - I completely understand, and if you read my post, I was simply "pointing out" that Webroot was detecting the latest stable as infected. I was just making you "aware" of the issue.

I do agree that it's probably a "False Positive" and I had already opened a "Trouble Ticket" with Webroot (earlier this morning, before you responded). I didn't want to open a "Trouble Ticket" without first discussing it with you.

I had already opened a Trouble Ticket (over an hour ago) prior to your last response.

<img width="814" alt="webroot_trouble_ticket" src="https://user-images.githubusercontent.com/133161/45493387-39c20880-b734-11e8-9679-0f792f358abc.png">
